src/v/archival/ntp_archiver_service.cc

```diff
@@ -394,4 +394,29 @@ ss::future<ntp_archiver::batch_result> ntp_archiver::upload_next_candidates(
       });
 }
 
+uint64_t ntp_archiver::estimate_backlog_size(cluster::partition_manager& pm) {
+    auto last_offset = _manifest.size() ? _manifest.get_last_offset()
+                                        : model::offset(0);
+    auto opt_log = pm.log(_manifest.get_ntp());
+    if (!opt_log) {
+        return 0U;
+    }
+    auto log = dynamic_cast<storage::disk_log_impl*>(opt_log->get_impl());
+    uint64_t total_size = 0;
+    uint64_t prev_size = 0;
+    // Ignore last segment
+    for (auto& segm : log->segments()) {
```

Comment:
maybe this will be easier to read when implemented with `std::accumulate(log->segments.begin(), std::prev(log->segments.end())` ?

Reply:
done